src/layout/index.tsx

```diff
@@ -35,8 +43,29 @@ interface ILayoutProps {
 	defaultSEO?: boolean
 	backgroundColor?: string
 	style?: React.CSSProperties
+	fullWidth?: boolean
 }
 
+const { chains, provider } = configureChains([mainnet, optimism], [publicProvider()])
```

Comment:
tbh since wallet functionality is only used in a few pages i think it would be better to add a new context for just those to avoid slowing down all the other pages because of the extra requests + code?

Reply:
need this to check subscription on other pages with csvdownload button

Reply:
why not use the auth token stored in localstorage?